datasource/keyvaultsecret/test-fixtures/template.pkr.hcl

```diff
@@ -0,0 +1,47 @@
+
+variable "keyvault_id" {
+    type=string
+}
+variable "client_id" {
+    type=string
+}
+variable "client_secret" {
+    type=string
+}
+variable "tenant_id" {
+    type=string
+}
+
+data "azure-keyvault-secret" "basic-example" {
+  name            = "packer-datasource-keyvault-test-secret"
+  keyvault_id     = "${var.keyvault_id}"
+  client_id       = "${var.client_id}"
+  client_secret   = "${var.client_secret}"
+  tenant_id       = "${var.tenant_id}"
```

Comment:
You can also just add a comment to this file telling that the `AZURE_*` envs are necessary to be set. If so you can remove the variables and these three lines, as well as their reference in `getAuthorizer()`

Reply:
I'm assuming azure consumes the env variables by default like AWS does. I think it does because I run my azure builds like this 🤔

Reply:
I tried Azure builder with only AZURE_* environment variable and it didn't work, it actually falls back to managed identity authentication, which obviously cannot work on my computer.
I'm genuinely curious to know how it works for you!
Having the same behavior than AWS plugin would be great indeed.

Reply:
The builder is setup to use managed identity if the user does not explicitly set subscription_id, client_id, and client_secret. Which is not configured automatically when using the AZURE_* environment variables. We run our builds and tests with the following variables:

```hcl
    "variables": {
        "client_id": "{{ env `ARM_CLIENT_ID` }}",
        "client_secret": "{{ env `ARM_CLIENT_SECRET` }}",
        "object_id": "{{ env `ARM_OBJECT_ID` }}",
        "subscription_id": "{{ env `ARM_SUBSCRIPTION_ID` }}",
        "tenant_id": "{{ env `ARM_TENANT_ID` }}",
        "resource_group_name": "{{ env `ARM_RESOURCE_GROUP_NAME` }}",
        "storage_account": "{{ env `ARM_STORAGE_ACCOUNT` }}"
    },
```

Reply:
Okay, this is way clearer! I'll push a change match this setup then :)

Reply:
👍 For the record this is an example in JSON. An HCL2 equivalent would be something like

```
variable "client_id" {
  type    = string
  default = "${env("ARM_CLIENT_ID")}"
}

variable "client_secret" {
  type    = string
  default = "${env("ARM_CLIENT_SECRET")}"
}

variable "object_id" {
  type    = string
  default = "${env("ARM_OBJECT_ID")}"
}

variable "resource_group_name" {
  type    = string
  default = "${env("ARM_RESOURCE_GROUP_NAME")}"
}

variable "storage_account" {
  type    = string
  default = "${env("ARM_STORAGE_ACCOUNT")}"
}

variable "subscription_id" {
  type    = string
  default = "${env("ARM_SUBSCRIPTION_ID")}"
}

variable "tenant_id" {
  type    = string
  default = "${env("ARM_TENANT_ID")}"
}```